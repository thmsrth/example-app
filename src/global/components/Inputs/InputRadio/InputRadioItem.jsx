import React from 'react';
import PropTypes from 'prop-types';

import { bemCls } from '../../../utils/ClassNameHelpers';

const InputRadioItem = ({
  className, name, label, value, checked, onChange, onBlur,
}) => (
  <div className={`InputRadioItem ${className}`}>
    <label
      className={`${bemCls(`InputRadioItem ${className}`, '__label')} ${
        bemCls(`InputRadioItem ${className}`, `__${name}__label`)}`}
    >
      <input
        type="radio"
        className={`${bemCls(`InputRadioItem ${className}`, '__input')} ${
          bemCls(`InputRadioItem ${className}`, `__${name}__input`)}`}
        name={name}
        value={value}
        checked={checked}
        onChange={onChange}
        onBlur={onBlur}
      />
      {label}
    </label>
  </div>
);

export default InputRadioItem;

InputRadioItem.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  className: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ]),
  checked: PropTypes.bool,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
};

InputRadioItem.defaultProps = {
  label: '',
  className: '',
  value: '',
  checked: false,
  onChange: () => {},
  onBlur: () => {},
};
