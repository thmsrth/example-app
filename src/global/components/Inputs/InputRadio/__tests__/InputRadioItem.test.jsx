import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import InputRadioItem from '../InputRadioItem';

const props = {
  className: 'MyInputRadioItem',
  name: 'type',
  label: 'My Radio Item Input Label',
  onChange: () => {},
  value: 'EMPLOYEE',
};

describe('InputRadioItem', () => {
  it('should render the radio input component correctly but not checked', () => {
    const tree = shallow(<InputRadioItem {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the radio input component correctly and checked', () => {
    const tree = shallow(<InputRadioItem {...props} checked />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
