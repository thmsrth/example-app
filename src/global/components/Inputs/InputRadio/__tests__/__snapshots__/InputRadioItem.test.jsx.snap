// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputRadioItem should render the radio input component correctly and checked 1`] = `
<div
  className="InputRadioItem MyInputRadioItem"
>
  <label
    className="InputRadioItem__label MyInputRadioItem__label InputRadioItem__type__label MyInputRadioItem__type__label"
  >
    <input
      checked={true}
      className="InputRadioItem__input MyInputRadioItem__input InputRadioItem__type__input MyInputRadioItem__type__input"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      type="radio"
      value="EMPLOYEE"
    />
    My Radio Item Input Label
  </label>
</div>
`;

exports[`InputRadioItem should render the radio input component correctly but not checked 1`] = `
<div
  className="InputRadioItem MyInputRadioItem"
>
  <label
    className="InputRadioItem__label MyInputRadioItem__label InputRadioItem__type__label MyInputRadioItem__type__label"
  >
    <input
      checked={false}
      className="InputRadioItem__input MyInputRadioItem__input InputRadioItem__type__input MyInputRadioItem__type__input"
      name="type"
      onBlur={[Function]}
      onChange={[Function]}
      type="radio"
      value="EMPLOYEE"
    />
    My Radio Item Input Label
  </label>
</div>
`;
